polylogue/mcp/server.py

```
@@ -251,294 +60,215 @@ def _conversation_to_dict(conv: Any) -> dict[str, Any]:
+        if provider:
+            filter_chain = filter_chain.provider(provider)
+        if since:
+            filter_chain = filter_chain.since(since)
 
-    filter_chain = ConversationFilter(repo)
-    filter_chain = filter_chain.contains("error")
+        conversations = filter_chain.list()
+        return json.dumps([_conversation_to_dict(c) for c in conversations], indent=2)
+
+    @mcp.tool()
+    def get_conversation(id: str) -> str:
```

Ruff (0.15.9)

[error] Function argument `id` is shadowing a Python builtin

(A002)